### ISSUE TITLE
Jaeger tracing is not enabled if [jaeger.tracing] is not present

### ISSUE THREAD

**Issue by Sayrus** [grafana/grafana#23626]
**What happened**:
Not having `[jaeger.tracing]` and the `address field` in my `grafana.ini` disables Jaeger Tracing.
According to the documentation:
```You can also use the standard JAEGER_* environment variables to configure Jaeger. ```
I configured Jaeger to not use an agent using only `JAEGER_ENDPOINT` and `JAEGER_SERVICE_NAME`.
The tracing was not enabled. I added the following to my `grafana.ini`:
```
[tracing.jaeger]
address = this-endpoint-does-not-exists
```
And now the tracing works. Note that `address` is bypassed by `JAEGER_ENDPOINT`.

**What you expected to happen**:
Having Jaeger environment variables should enable tracing without needing to add a tracing.jaeger section.

**How to reproduce it (as minimally and precisely as possible)**:
Start Grafana with `JAEGER_ENDPOINT` (or I'd guess it would work with `JAEGER_AGENT_HOST` and `JAEGER_AGENT_PORT`) and `JAEGER_SERVICE_NAME`. Tracing won't be enabled.
Add a tracing.jaeger section and a placeholder value in address (as it is overridden by both `JAEGER_AGENT_HOST` and `JAEGER_ENDPOINT`) and tracing will now work as expected. 

**Anything else we need to know?**:
I do think this behavior comes from `parseSettings`.
https://github.com/grafana/grafana/blob/66b70b50b809bab6c0f4e1e148af60a141261769/pkg/infra/tracing/tracing.go#L47

`ts.enabled` is set to true if, and only if, a non empty address was set. And `initGlobalTracer` which in turn calls `initJaegerCfg` is only called if `ts.enabled` is true.

**Environment**:
- Grafana version: 6.7.1 
- Data source type & version: N/A
- OS Grafana is installed on: Kubernetes / Centos
- User OS & Browser: N/A
- Grafana plugins: N/A
- Others:

**Comment by bergquist** [comment]
Thank you for reporting this.

You can work around this by overriding the address setting using environment variables in Grafana. https://grafana.com/docs/grafana/latest/installation/configuration/#configure-with-environment-variables

```
GF_TRACING_JAEGER_ADDRESS="localhost:6831"
```
I understand that this is frustrating but I'm not sure it's a high prio. Leaving it open in case more people swing by and upvote it.

**Comment by bergquist** [comment]
I'm closing this due to inactivity. If this gets more attention we can reopen it again.

**Comment by carlosjgp** [comment]
This is still the case on Grafana `7.0.5`

to be able to enable traces I did

Using Grafana Helm chart

```yaml
env:
  JAEGER_AGENT_HOST: simplest-agent
  JAEGER_AGENT_PORT: 6831
  JAEGER_SERVICE_NAME: grafana
```

but that didn't work. Then I added

```yaml
grafana.ini:
  tracing.jaeger:
    address: fake-address-that-is-overriden
```

And that configured Grafana to send traces to Jaeger

Originally tested with Grafana 7.0.3 but there was a problem with the client
https://github.com/grafana/grafana/issues/25563
https://github.com/grafana/grafana/commit/e65f4c4a76d9a38c67961ed01252c5cb2b04ff0e

Fixed on `7.0.5`
